src/main/java/io/pravega/connectors/hadoop/PravegaOutputFormat.java

```diff
@@ -123,10 +126,25 @@ protected EventStreamClientFactory getClientFactory(String scope, URI controller
         if (router != null && !PravegaEventRouter.class.isAssignableFrom(router.getClass())) {
             throw new IOException(router.getClass() + " is not a type of PravegaEventRouter");
         }
+
+        PravegaClientConfig pravegaClientConfig = PravegaClientConfig.fromDefaults();
```

Comment:
As mentioned earlier, there is a duplication of this logic. Please abstract this out and reuse.

Reply:
reorganized code to use helper class to derive the configuration instance